src/platform/datapath_raw_socket.c

```diff
@@ -97,7 +96,7 @@ CxPlatTryAddSocket(
     Entry = CxPlatHashtableLookup(&Pool->Sockets, Socket->LocalAddress.Ipv4.sin_port, &Context);
     while (Entry != NULL) {
         CXPLAT_SOCKET* Temp = CONTAINING_RECORD(Entry, CXPLAT_SOCKET, Entry);
-        if (QuicAddrCompareIp(&Temp->LocalAddress, &Socket->LocalAddress)) {
+        if (!CxPlatSocketCompare(Temp, &Socket->LocalAddress, &Socket->RemoteAddress)) {
```

Comment:
Why is this negated? Isn't it a collision if CxPlatSocketCompare returns TRUE?

Reply:
I agree, it's a bug.
```suggestion
        if (CxPlatSocketCompare(Temp, &Socket->LocalAddress, &Socket->RemoteAddress)) {
```